### DISCUSSION TITLE
How to get type of field name

### DISCUSSION THREAD

**Discussion by mbibko** [react-hook-form/react-hook-form#8989] [Q&A]
I register new input field with useController and literal type, i.e. InputEnum.example.
```
enum InputEnum {
  example = "example"
}

const Input = ({ name }: { name: InputEnum.example }) => {
  const { control } = useFormContext<{ [name]: string }>();
  const { field } = useController({
    name,
    control,
    rules: { required: "This is required message" }
  });
  return <input {...field} />;
};
```
It's work fine in the same file. But how can I get correct type if I register input in separate file? I get incorrect type here https://codesandbox.io/s/react-hook-literal-type-3h13h5?file=/src/Input2.tsx

`A computed property name in a type literal must refer to an expression whose type is a literal type`

Where I wrong? Should I get field type from generic type or there are another way to get it?
Thanks for help!

**Comment by Moshyfawn** [comment]
RHF doesn't support `Enum`s as field values #8898
